scrapers/scrapers/spiders/mo-bills.py

```diff
@@ -0,0 +1,687 @@
+                summary_name, path, media_type=mimetype, on_duplicate="ignore"
+            )
+
+        # house fiscal note
+        for row in response.xpath('//BillInformation/FiscalNote'):
+            path = row.xpath('./FiscalNoteLink/text()').get().strip()
+            version = path.split('/')[-1].replace('.pdf', '')
+            summary_name = f'Fiscal Note {version}'
+            if ".pdf" in path:
+                mimetype = "application/pdf"
+            else:
+                mimetype = ""
+            bill.add_document_link(
+                summary_name, path, media_type=mimetype, on_duplicate="ignore"
+            )
+
+    # SENATE BILL
+    def parse_senate_billpage(self, response):
+        # get all the info needed to record the bill
+        # TODO probably still needs to be fixed
+        bill_url = response.url
+        bill_id = response.xpath('//*[@id="lblBillNum"]//text()').get()
+        bill_title = response.xpath('//*[@id="lblBillTitle"]//text()').get()
+        bill_desc = response.xpath('//*[@id="lblBriefDesc"]//text()').get()
+        # bill_lr = response.xpath('//*[@id="lblLRNum"]/text()').get()
+
+        bill_type = "bill"
+        triplet = bill_id[:3]
+        if triplet in bill_types:
+            bill_type = bill_types[triplet]
+
+        subs = []
+        bid = bill_id.replace(" ", "")
+
+        if bid in self.subjects:
+            subs = self.subjects[bid]
+
+        if bid == "XXXXXX":
+            self.logger.warning(f"Skipping Junk Bill {bid}")
+            return
+
+        bill = Bill(
+            bill_id,
+            title=bill_desc,
+            chamber="upper",
+            legislative_session=self.session,
+            classification=bill_type,
+        )
+        bill.subject = subs
+        bill.add_abstract(bill_desc, note="abstract")
+        bill.add_source(bill_url)
+
+        if bill_title:
+            bill.add_title(bill_title)
+
+        # Get the primary sponsor
+        bill_sponsor = response.xpath(
+            '//a[@id="hlSponsor"]//text()').get() or response.xpath('//span[@id="lSponsor"]//text()').get()
+
+        # bill_sponsor_link = sponsor.attrib.get('href')
+        bill.add_sponsorship(
+            bill_sponsor, entity_type="person", classification="primary", primary=True
+        )
+
+        amendment_links = response.xpath(
+            '//a[contains(@href,"ShowAmendment.asp")]')
+        for link in amendment_links:
+            link_text = link.xpath(".//text()").get().strip()
+            if not link_text:
+                link_text = "Missing description"
+            if "adopted" in link_text.lower():
+                link_url = link.xpath("@href").get()
+                bill.add_version_link(
+                    link_text,
+                    link_url,
+                    media_type="application/pdf",
+                    on_duplicate="ignore",
+                )
+
+        # cosponsors show up on their own page, if they exist
```

Comment:
Since we are chaining a series of requests (first to get co-sponsors, then to get actions, then to get versions, then to get fiscal notes, then finally to yield the bill), I think would be worth adding a comment here that describes this chain of requests that is going to happen, just to make it clearer to other devs

Reply:
I've fixed it.